### PR TITLE
docs: add sitemap overview

### PR DESCRIPTION
**Pull request by bc-andreadao**
## What/Why?

[DEVDOCS-6011](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6011)

Add doc for XML sitemap

This PR relates to [PR 20](https://github.com/bigcommerce/catalyst.dev/pull/20) in catatlyst.dev (for menu changes)

## Testing
N / A

[DEVDOCS-6011]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6011?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/sitemap.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Sitemap
+
+The `/sitemap.xml` URL provides the complete BigCommerce sitemap, which includes canonical URLs for all channels associated with Catalyst.
```
Comment (bc-andreadao) [inline docs/sitemap.md:3]: What domain does the sitemap have?

File: docs/sitemap.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+These URLs correspond to the [permanent URLs](/docs/storefront/graphql#i-want-to-run-requests-in-the-context-of-another-channel) used when making requests to the GraphQL Storefront API. The URLs correspond to a specific channel, and when accessed, returns the relevant sitemap for that channel.
```
Comment (bc-andreadao) [inline docs/sitemap.md:4]: Just confirming, are canonical URLs the same as permanent URLs? Are custom domains supported?

File: docs/sitemap.md (hunk 1 above)
Comment (chanceaclark) [reply docs/sitemap.md:3]: The top level sitemap index has the canonical URLs, but the individual sitemaps have the site URL.

## Sitemap Index
Screenshot 2024-08-29 at 13 56 22

## Individual Sitemap
Screenshot 2024-08-29 at 13 56 56

File: docs/sitemap.md (hunk 2 above)
Comment (chanceaclark) [reply docs/sitemap.md:4]: Canonical does equal permanent URLs. 

>Are custom domains supported?

Not sure what you mean here? Maybe my previous comment answers this?

Comment (bc-andreadao) [reply docs/sitemap.md:4]: Yeah, I think so 👍

**Comment by bc-andreadao** [comment]
Thanks @chanceaclark. If everything else looks good, an approval would be great

**Review by chanceaclark — APPROVED** [review]

**Review by jorgemoya — APPROVED** [review]

**Pull request merged**